ui/src/pages/publications/[id]/index.tsx

```diff
@@ -58,15 +58,18 @@ export const getServerSideProps: Types.GetServerSideProps = async (context) => {
 
     // fetch data concurrently
     const promises: [
-        Promise<Interfaces.PublicationVersion | void>,
+        Promise<AxiosResponse<Interfaces.PublicationVersion> | void>,
         Promise<Interfaces.BookmarkedEntityData[] | void>,
         Promise<Interfaces.PublicationWithLinks | void>,
         Promise<Interfaces.Flag[] | void>,
         Promise<Interfaces.BaseTopic[] | void>
     ] = [
         api
-            .get(`${Config.endpoints.publications}/${requestedId}/publication-versions/latest`, token)
-            .then((res) => res.data)
+            .get(`${Config.endpoints.publications}/${requestedId}/publication-versions/latest`, token, {
+                // We need to know the return code of this request if it is 403/404 so we can redirect,
+                // so this stops it from being caught and not checkable in the following code.
+                validateStatus: (status) => [403, 404].includes(status) || (status >= 200 && status < 300)
```

Comment:
I recommend using the .catch block and return the status code and error message from there if needed, instead of using "validateStatus".